plugins/default/lib/component_health.py

```diff
@@ -79,22 +75,25 @@ def report(summary, test_results):
         print 'Test: %s - Status: %s - Details: %s' % (test['description'], test['test_status'], test['result'])
 
 
-def main():
+def check(host, port, endpoint):
     try:
-        args = generate_parser().parse_args()
-        host = args.host
-        port = args.port
-        endpoint = args.endpoint
         response = do_request(host, port, endpoint)
         nagios_status, test_summary, test_results = parse_response(response)
         report(test_summary, test_results)
     except RequestError as error:
         print error.message
+        # REVIEW: shouldn't this be nagios.UNKNOWN? If the request failed, we
+        # "don't know" if the component is healthy or not.
```

Comment:
> Since we have full control of all of the containers/services in our RHMAP projects, every check should be capable of determining an exact status without encountering UNKNOWN.

This is the case when there was a request error -- e.g. the service returned something unexpected.

For now, see the line below, we're returning `nagios.CRIT` in that case, but I suspect it would be more appropriate to say `nagios.UNKNOWN`, because a request error means there was an error performing the check.

Do you agree?